core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -145,32 +161,27 @@ class Analyzer(
   def startup: Receive = withLabel("startup") {
     case FullTypeCheckCompleteEvent =>
       reporter.enable()
-      // legacy clients expect to see AnalyzerReady and a
-      // FullTypeCheckCompleteEvent on connection.
-      broadcaster ! Broadcaster.Persist(AnalyzerReadyEvent)
-      broadcaster ! Broadcaster.Persist(FullTypeCheckCompleteEvent)
       context.become(ready)
       unstashAll()
-
     case other =>
       stash()
   }
 
   def ready: Receive = withLabel("ready") {
     case FullTypeCheckCompleteEvent =>
       broadcaster ! FullTypeCheckCompleteEvent
-
     case RestartScalaCompilerReq(id, strategy) =>
-      restartCompiler(strategy, id)
-
+      restartCompiler(strategy)
+    case SuspendAnalyzer =>
+      loadedFiles = scalaCompiler.loadedFiles // remember the state
+      scalaCompiler.askShutdown()
+      context.become(suspended)
     case UnloadAllReq =>
-      restartCompiler(ReloadStrategy.UnloadAll, None)
+      restartCompiler(ReloadStrategy.UnloadAll)
       sender ! VoidResponse
-
     case TypecheckModule(id) =>
-      restartCompiler(ReloadStrategy.LoadProject, Some(id))
+      restartCompiler(ReloadStrategy.LoadProject)
       sender ! VoidResponse
-
     case req: RpcAnalyserRequest =>
```

Comment:
this means that none of the messages above will cancel the 5 minute timebomb... but since they are not direct user requests I think that's ok.